biomage/experiment/download.py

```diff
@@ -0,0 +1,338 @@
+        raise Exception("No data returned from query")
+
+    samples = json.loads(json_text)
+
+    # Create a dictionary of samples
+    result = {}
+
+    for sample in samples:
+
+        if not result.get(sample["sample_name"]):
+            result[sample["sample_name"]] = []
+
+        result[sample["sample_name"]].append(sample)
+
+    return result
+
+
+def _get_samples_v2(experiment_id, input_env):
+    """
+    Get samples data for v2
+    """
+    SANDBOX_ID = "default"
+    REGION = "eu-west-1"
+    USER = "dev_role"
+
+    command = f"""psql -c "SELECT json_agg(t) FROM ( \
+        SELECT sample_file_id , sample_id, s3_path, name AS sample_name FROM ( \
+            SELECT sample_file_id, sample_id, s3_path FROM ( \
+                SELECT  sample_file_id, sample_id \
+                    FROM sample_to_sample_file_map \
+                    WHERE sample_id \
+                        IN ( \
+                        SELECT id \
+                        FROM sample \
+                        WHERE experiment_id = '{experiment_id}' \
+                    ) \
+            ) AS a \
+            LEFT JOIN sample_file sf ON a.sample_file_id = sf.id \
+        ) AS b \
+        INNER JOIN sample ON b.sample_id = sample.id ) as t"\
+    """
+
+    print(f"Querying samples for {experiment_id}...")
+
+    result_str = run_rds_command(command, SANDBOX_ID, input_env, USER, REGION, True)
+    return _process_query_output(result_str)
+
+
+def _download_samples_v2(experiment_id, input_env, output_path, name_sample_using_id):
+    """
+    Download samples associated with an experiment from a given environment for v2.\n
+    """
+    bucket = f"{SAMPLES_BUCKET}-{input_env}"
+
+    samples_list = _get_samples_v2(experiment_id, input_env)
+    num_samples = len(samples_list)
+
+    print(f"{num_samples} samples found. Downloading sample files...\n")
+
+    for sample_idx, value in enumerate(samples_list.items()):
+        sample_name, samples = value
+
+        if name_sample_using_id:
+            sample_name = samples[0]["sample_id"]
+
+        num_files = len(samples)
+
+        print(
+            f"Downloading files for sample {sample_name} (sample {sample_idx+1}/{num_samples})",
+        )
+
+        for file_idx, sample in enumerate(samples):
+
+            s3_path = sample["s3_path"]
+
+            file_name = Path(s3_path).name
+            file_path = output_path / sample_name / file_name
+
+            print(
+                f"> Downloading {sample_name}/{file_name} (file {file_idx+1}/{num_files})"
+            )
+
+            s3client = boto3.client("s3")
+            s3client.head_object(Bucket=bucket, Key=s3_path)
+            _download_file(bucket, s3_path, file_path)
+
+        print(f"Sample {sample_name} downloaded.\n")
+
+    _create_sample_mapping(samples_list, output_path)
+    print("All samples for the experiment have been downloaded.")
+
+
+def _download_rds(experiment_id, input_env, output_path, processed=False):
```

Comment:
could you rename to `download_rds_files`? Just to disambiguate from the other rds

Reply:
I'll rename it `download_seurat_files`

Reply:
well, technically before they are processed they are not Seurat files. They are matrices, barcodes, features files. Also, what if we added support for non Seurat files too? Like scanpy? Then this rename becomes out of date

Reply:
Ah right, you have a point. I was afraid "processed_file" is ambiguous, as a "file" can mean a lot of things.

Reply:
Ah right, you have a point. Then renaming it `download_rds_files` is not good too.

Reply:
But I think it's fine for now, that's another matter to worry about